21 anpassen der spawnmonster klasse

Manuel1598 opened the pull request:
Die `SpawnMonsterHandler`-Klasse wurde aktualisiert, um die Zone direkt aus der eingehenden Nachricht zu extrahieren und bei der Monstererzeugung zu verwenden. Diese Änderung erleichtert die clientseitige Implementierung, da alle notwendigen Daten und Informationen in einer einzigen Response zusammengefasst werden, was das Erstellen von `MonsterDTO` Objekten vereinfacht.

Zusätzlich wurden die zugehörigen Tests angepasst, um die korrekte Übergabe und Handhabung der Zone zu überprüfen. Dies stellt sicher, dass die `spawnRandomMonster`-Methode korrekt funktioniert und die Tests unter den neuen Anforderungen erfolgreich durchlaufen werden.

Closes #21

Manuel1598 committed:
feat: Implementation and customisation of the SpawnMonsterHandler class

Manuel1598 committed:
fix: Change the transfer of zone and ring

-These were initially calculated with modulo 4, but this is not possible as we have 4 zones

Manuel1598 committed:
merge: Integrate latest changes from main into current branch

Manuel1598 committed:
feat: Enhance SpawnMonsterHandler with updated tests and public access for spawning method

paulxyh committed:
feat: Testing for UtilityMethods added

paulxyh committed:
fix: CodeCleanup UtilityMethodsTest

Manuel1598 committed:
fix: Remove duplicate increment of monsterId in SpawnMonsterHandler

Manuel1598 committed:
refactor: Enhance WebSocket message handling in GameHandler

- Introduce a centralized sendMessage method to standardize JSON message construction and sending.
- Ensure that all outgoing messages include a clear type designation for easier client-side processing.

Manuel1598 committed:
feat: Adding a playerRollsDiceHandler

-Serves to ensure that only one player at a time plays the dice.

Manuel1598 committed:
feat: Adding a GameRoundHandler

-ensures that the counting of rounds works correctly for all players

Manuel1598 committed:
feat: Send spawn monster message to all players in the lobby

-Modified the SpawnMonsterHandler to broadcast the spawn monster message to all players' sessions in the lobby instead of sending it to just the current session.

paulxyh commented on src/main/java/at/aau/se2/model/Monster.java:
```diff
@@ -12,8 +12,8 @@ public abstract class Monster extends Tower implements JsonSerializable {
         super(id);
-        this.zone = zone % 4; // 0-3
-        this.ring = ring % 4; // 0-3
+        this.zone = zone;
```
`% 4` ist nochmal eine extra Überprüfung, dass wirklich nur 0-3 als Wert gesetzt werden kann. Die ist dafür da, dass wir im Client keinen Fehler geschmissen bekommen

Manuel1598 replied:
Ich gebe vom Client die Werte 1-4 an den Server für Zone 1-4 deswegen habe ich das müssen rausnehmen, da er bei Zone 4 dann 0 draus macht.

paulxyh committed:
refactor: merged switch-crad-with-player

paulxyh committed:
refactor: Code Cleanup

paulxyh committed:
refactor: SecureRandom Objekt als Klassenattribut angelegt

paulxyh committed:
refactor: CodeCleanup and ensuring the Username can only be set once per session

paulxyh committed:
refactor: CodeCleanup

paulxyh committed:
refactor: SecureRandom as parameter of the Constructor

paulxyh committed:
feat: Testing for Player.java added

Manuel1598 committed:
test: Add DrawCardHandler test class

Implement unit tests for the DrawCardHandler class to ensure its functionality, covering the drawRandomCard method and JSON conversion functionality.

Manuel1598 committed:
Merge remote-tracking branch 'origin/23-testing' into 21-anpassen-der-spawnmonster-klasse

# Conflicts:
#	src/main/java/at/aau/se2/handler/game/GameHandler.java

Manuel1598 committed:
merge: Merge branch testing into this

Manuel1598 committed:
feat: Add turn rotation logic to TurnHandler to manage player actions based on the game state

Manuel1598 committed:
refactor: TurnHandler to use a fixed player count and update turn notification

- Remove dependency on Lobby object to determine player count and simplify class design.
- Change turn notification to send player index instead of username to align with client-side handling.

Manuel1598 committed:
refactor: Implement player notification across all clients when the turn changes

Manuel1598 committed:
Merge branch 'refs/heads/main' into 21-anpassen-der-spawnmonster-klasse

# Conflicts:
#	src/main/java/at/aau/se2/handler/game/GameHandler.java
#	src/main/java/at/aau/se2/handler/game/subhandlers/DrawCardHandler.java
#	src/main/java/at/aau/se2/handler/game/subhandlers/RegisterUsernameHandler.java
#	src/main/java/at/aau/se2/handler/game/subhandlers/RequestUsernamesForSwitchHandler.java
#	src/main/java/at/aau/se2/handler/game/subhandlers/SwitchCardDeckHandler.java
#	src/main/java/at/aau/se2/handler/game/subhandlers/SwitchCardPlayerHandler.java
#	src/main/java/at/aau/se2/utils/Player.java

Manuel1598 committed:
refactor: Changed name of Method

Manuel1598 committed:
refactor: Changed name of Method again

Manuel1598 committed:
refactor: refactored TestClasses in correct packages

Manuel1598 committed:
Merge remote-tracking branch 'origin/21-anpassen-der-spawnmonster-klasse' into 21-anpassen-der-spawnmonster-klasse
